src/components/pages/LoginPage/LoginUPort.js

```diff
@@ -2,17 +2,37 @@ import React, { Component } from 'react'
 import styles from './styles'
 import { connect } from 'react-redux'
 import { RaisedButton } from 'material-ui'
+import web3Provider from '../../../network/Web3Provider'
+import uportProvider, { decodeMNIDaddress } from '../../../network/UportProvider'
+import { addError, clearErrors, loadAccounts, selectAccount } from '../../../redux/network/actions'
 
-const mapStateToProps = (state) => ({})
+const mapStateToProps = (state) => ({
+  accounts: state.get('network').accounts
+})
 
 const mapDispatchToProps = (dispatch) => ({
-  // TODO @dkchv
+  loadAccounts: () => dispatch(loadAccounts()),
+  selectAccount: (account) => dispatch(selectAccount(account)),
+  addError: (error) => dispatch(addError(error)),
+  clearErrors: () => dispatch(clearErrors())
 })
 
 @connect(mapStateToProps, mapDispatchToProps)
 class LoginUPort extends Component {
   handleLoginClick = () => {
-    // TODO @dkchv
+    this.props.clearErrors()
+    web3Provider.setWeb3(uportProvider.getWeb3())
+    web3Provider.setProvider(uportProvider.getProvider())
+    this.props.loadAccounts().then(() => {
+      if (!this.props.accounts) {
+        return
+      }
+      const { address } = decodeMNIDaddress(this.props.accounts[0])
+      this.props.selectAccount(address)
+      this.props.onLogin()
+    }).catch(e => {
+      this.props.addError(e.message)
+    })
```

Comment:
whole logic of this handler should be extracted into the separate action creator